juju/juju
# Optional k8s charm pv

### Checklist


There's a change to api parameters but it's for a controller facade.

----

## Description of change

A k8s charm can now signal that it does not require a PV to be created for charm state storage. It does this by setting min-juju-version=2.8.0. Existing charms which do require PV storage are still catered for, and if a charm is upgraded, any PV is retained.

The first commit factors out the min version check to a common function. It has been tweaked to allow a charm specifying a major.minor.0 version to work with a beta/rc (to allow testing).

## QA steps

Deploy a k8s charm
Edit the charm to add min-juju-version=2.8.0
upgrade the charm and check the PV and PVC are still used
deploy the edited charm again as a different application
check that no operator PV is created
